include/app/drivers/blink.h

```diff
@@ -82,10 +83,11 @@ __syscall int blink_set_period_ms(const struct device *dev,
 static inline int z_impl_blink_set_period_ms(const struct device *dev,
 					     unsigned int period_ms)
 {
-	const struct blink_driver_api *api =
-		(const struct blink_driver_api *)dev->api;
+	CHECKIF(!DEVICE_API_IS(blink, dev)) {
+		return -ENODEV;
+	}
```

Comment:
if true, this is a programming error and so an assert/panic is a better choice. returning an error for such cases is useless.

Reply:
I know, but only assert makes sense.

Reply:
ie, the compiler would fail if we had strong typing, so runtime checks should also panic always.